pkg/controller/builder.go

```diff
@@ -247,6 +252,62 @@ func UpdateDeploymentProxyHook(pc clients.ProxyClientInterface) deploymentcontro
 		if len(errs) > 0 {
 			return errors.Join(errs...)
 		}
+
+		return nil
+	}
+}
+
+func UpdateDeploymentFeatureGatesHook(
+	featuresAccessor featuregates.FeatureGateAccess,
+	featuresMapper internalfeatures.MapperInterface,
+) deploymentcontroller.DeploymentHookFunc {
+	return func(_ *operatorv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		klog.FromContext(context.Background()).WithName("builder").V(0).Info("FeatureGatesHook updating environment", "deployment", deployment.Name)
+
+		featuregates, err := featuresAccessor.CurrentFeatureGates()
```

Comment:
Is it possible to iterate `featuregates`? My hope is that this hook code would not have to change whenever we add/remove downstream feature gates.

I was thinking something like:
```
var upstreamGates []string
for _, fg := range featuregates {
    // featuresMapper.UpstreamForDownstream(fg) returns nil if we
    // don't have a mapping for the downstream feature (we'll see _all_
    // of the OCP feature gates, but we only care about ours)
    upstreamGates = append(upstreamGates, featuresMapper.UpstreamForDownstream(fg))
}
```

Reply:
that's a good idea and I think we should be able to leverage `Mapper` to provide a list of downstream features that are of interest here. Mapper has that information already (and it was meant to have it) and this way we should be able to avoid having to loop through all features.